staging/src/k8s.io/kube-scheduler/config/v1alpha2/types.go

```diff
@@ -99,6 +101,20 @@ type KubeSchedulerConfiguration struct {
 	Extenders []v1.Extender `json:"extenders"`
 }
 
+// DecodeNestedObjects decodes plugin args for known types.
+func (c *KubeSchedulerConfiguration) DecodeNestedObjects(d runtime.Decoder) error {
```

Comment:
do we need a symmetric EncodeNestedObjects so we can round-trip a scheduler config from v1alpha2 -> internal -> v1alpha2 and the args convert and get marshaled correctly?

Reply:
We don't need conversion from internal to v1alpha2. But I added test to encode from v1alpha2

Reply:
> We don't need conversion from internal to v1alpha2

Are you sure? Does the following work:
1. decode config file with args to internal KubeSchedulerConfiguration (args end up as internal Args objects)
2. encode that internal KubeSchedulerConfiguration object to v1alpha2 (args should get converted back to v1alpha2 and serialized correctly)

Reply:
With the code as-is, that doesn't work (the nested types are directly encoded from internal, thus get a starting Uppercase). We would need to implement the interface.

Reply:
I realized that, with the current implementation, v1alpha2 args don't get gvk, which might be dangerous.
But now there is an issue, because the nested objects get enconded into YAML before
https://github.com/kubernetes/kubernetes/blob/470dfbfc4848cee4897a1b176d20611668820492/staging/src/k8s.io/apimachinery/pkg/runtime/serializer/json/json.go#L305-L314